service/controller/chart/v1/resource/releasemigration/resource.go

```diff
@@ -53,3 +70,103 @@ func New(config Config) (*Resource, error) {
 func (r *Resource) Name() string {
 	return Name
 }
+
+func (r *Resource) findHelmV2Releases(ctx context.Context) ([]string, error) {
+	lo := metav1.ListOptions{
+		LabelSelector: fmt.Sprintf("%s=%s", "OWNER", "TILLER"),
+	}
+
+	// Check whether helm 2 release configMaps still exist.
+	cms, err := r.k8sClient.CoreV1().ConfigMaps(r.tillerNamespace).List(lo)
+	if err != nil {
+		return nil, microerror.Mask(err)
+	}
+
+	hadReleases := map[string]bool{}
+	for _, cm := range cms.Items {
+		name, _ := cm.GetLabels()["NAME"]
+		if _, ok := hadReleases[name]; !ok {
+			hadReleases[name] = true
+		}
+	}
+
+	releases := make([]string, 0, len(hadReleases))
+	for k := range hadReleases {
+		releases = append(releases, k)
+	}
+
+	return releases, nil
+}
+
+func (r *Resource) ensureReleasesMigrated(ctx context.Context) error {
+	// 0. Found all dangling helm release v2
```

Comment:
Do these comments help? I think so; would like to preserve it.

Reply:
Agree but the numbering is not our usual comments style.